OCCommunicationLib/OCCommunicationLib/OCOAuth2Authentication/OCOAuth2Manager.m

```diff
@@ -0,0 +1,134 @@
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#import "OCOAuth2Manager.h"
+
+@implementation OCOAuth2Manager
+
++ (void) getAuthDataByOAuth2Configuration:(OCOAuth2Configuration *)oauth2Configuration
+                          withBaseURL:(NSString *)baseURL
+                             refreshToken:(NSString *)refreshToken
+                                userAgent:(NSString *)userAgent
+                                  success:(void(^)(OCCredentialsDto *userCredDto))success
+                                  failure:(void(^)(NSError *error))failure {
+    
+    [UtilsFramework deleteAllCookies];
+    
+    [self refreshTokenAuthRequestByOAuth2Configuration:oauth2Configuration withBaseURL:baseURL refreshToken:refreshToken userAgent:userAgent
+     
+    success:^(NSHTTPURLResponse *response, NSError *error, NSData *data) {
+        
+        NSDictionary *dictJSON;
+        OCCredentialsDto *userCredDto = [OCCredentialsDto new];
+        
+        if (data != nil) {
+            
+            NSError *errorJSON = nil;
+            NSLog(@"data = %@",[[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding]);
+            dictJSON = [NSJSONSerialization JSONObjectWithData:data
+                                                       options:kNilOptions
+                                                         error:&errorJSON];
+            if (errorJSON == nil) {
+                if (dictJSON[@"error"] != nil && ![dictJSON[@"error"] isEqual:[NSNull null]] ) {
+                    
+                    NSString *message = (NSString*)[dictJSON objectForKey:@"error"];
+                    
+                    if ([message isKindOfClass:[NSNull class]]) {
+                        message = @"";
+                    }
+                    
+                    NSError *error = [UtilsFramework getErrorWithCode:response.statusCode andCustomMessageFromTheServer:message];
```

Comment:
Should better use the other method in ```UtilsFramework``` to get ```NSError```s, ```getErrorById```, with one of the new ```OCErrorOAuth2xxxx```constants as id. The value of the "error" element in the JSON response is not appropriate for user messages.

Reply:
Agree, let's changed with a default one